app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java

```diff
@@ -2559,7 +2548,10 @@ public Mono<Boolean> isRepoLimitReached(String workspaceId, Boolean isClearCache
                     }
                     return this.getApplicationCountWithPrivateRepo(workspaceId)
                             .map(privateRepoCount -> {
-                                if (limit > privateRepoCount) {
+                                if (limit >= privateRepoCount) {
+                                    if (isClearCache) {
```

Comment:
```suggestion
                                if (limit >= privateRepoCount && isClearCache) {
```

Without knowledge of anything in the code flow, this is minor nit I just noticed. Feel free to ignore if this is irrelevant. 😄